agent/consul/state/peering.go

```diff
@@ -852,18 +865,24 @@ func exportedServicesForPeerTxn(
 		}
 		info.Protocol = protocol
 
-		if dc != "" && !structs.IsProtocolHTTPLike(protocol) {
-			// We only need to populate the targets for replication purposes for L4 protocols, which
-			// do not ultimately get intercepted by the mesh gateways.
-			idx, targets, err := discoveryChainOriginalTargetsTxn(tx, ws, dc, svc.Name, &svc.EnterpriseMeta)
-			if err != nil {
-				return fmt.Errorf("failed to get discovery chain targets for service %q: %w", svc, err)
-			}
+		idx, targets, err := discoveryChainOriginalTargetsTxn(tx, ws, dc, svc.Name, &svc.EnterpriseMeta)
+		if err != nil {
+			return fmt.Errorf("failed to get discovery chain targets for service %q: %w", svc, err)
+		}
+		if idx > maxIdx {
+			maxIdx = idx
+		}
```

Comment:
Does moving this out of the check for L4 protocols change the semantics of when we update the maxIdx? Seems like we might want to wait to increment until we decide if the targets are valid.

Reply:
Ah, I see what you mean. So maybe move the `if idx > maxIdx{` until right before the return for the function?